Gopkg.toml

```diff
@@ -0,0 +1,74 @@
+# Gopkg.toml example
+#
+# Refer to https://golang.github.io/dep/docs/Gopkg.toml.html
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
+# ignored = ["github.com/user/project/pkgX", "bitbucket.org/user/project/pkgA/pkgY"]
+#
+# [[constraint]]
+#   name = "github.com/user/project"
+#   version = "1.0.0"
+#
+# [[constraint]]
+#   name = "github.com/user/project2"
+#   branch = "dev"
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#   name = "github.com/x/y"
+#   version = "2.4.0"
+#
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
+
+
+[[constraint]]
+  name = "github.com/dgraph-io/badger"
+  version = "1.5.4"
+
+[[constraint]]
+  name = "github.com/golang/snappy"
+  version = "0.0.1"
+
+[[constraint]]
+  name = "github.com/ipfs/go-datastore"
+  version = "0.0.1"
+
+[[constraint]]
+  name = "github.com/ipfs/go-ipfs-addr"
+  version = "0.0.1"
+
+[[constraint]]
+  name = "github.com/libp2p/go-libp2p"
+  version = "0.0.2"
+
+[[constraint]]
+  name = "github.com/libp2p/go-libp2p-host"
+  version = "0.0.2"
+
+[[constraint]]
+  name = "github.com/libp2p/go-libp2p-kad-dht"
+  version = "0.0.5"
+
+[[constraint]]
+  name = "github.com/libp2p/go-libp2p-net"
+  version = "0.0.1"
+
+[[constraint]]
+  name = "github.com/libp2p/go-libp2p-peer"
+  version = "0.1.0"
+
+[[constraint]]
+  name = "github.com/libp2p/go-libp2p-peerstore"
+  version = "0.0.1"
+
+[[constraint]]
+  name = "github.com/multiformats/go-multiaddr"
+  version = "0.0.2"
+
```

Comment:
do we need to include every dependency we have here? if so there are a few more we should add

Reply:
these are the dependencies that currently exist in the codebase, you only add the dependency when it's being used. For the p2p dependencies, they will be included when you use `dep ensure` or `dep ensure -add` on your own branch where the dependencies are being used

Reply:
i see, so on my p2p PR, i should add it to the toml file? (once this gets merged)

Reply:
okay i see, that makes sense thanks